Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -260,14 +280,17 @@ public sealed class Server : ServerBase, IDisposable {
             }
 
             await _analyzer.ReloadModulesAsync();
-
             // re-analyze all of the modules when we get a new set of modules loaded...
             foreach (var entry in _analyzer.ModulesByFilename) {
                 _queue.Enqueue(entry.Value.ProjectEntry, AnalysisPriority.Normal);
             }
         }
 
         public override Task<CompletionList> Completion(CompletionParams @params) {
+            IfTestWaitForAnalysisComplete();
+            // Make sure document is enqueued for processing
+            _documentChangeProcessingComplete.Wait(200, CancellationToken);
```

Comment:
Shouldn't we wait for this before waiting for analysis? And shouldn't we wait forever if this is necessary? (I assume most of the time there will be no wait actually happen here.)

Reply:
We're also likely better off marking the entry as parse pending as soon as possible when a change comes in, and then the WaitForCurrentParse below will do the correct wait.